gunfolds/utils/clingo.py

```diff
@@ -41,26 +41,45 @@ def run_clingo(command,
         trendy: Use defaults geared towards industrial problems
     :type configuration: string
 
-    :param cpath: clingo path
-    :type cpath: string
 
     :param pnum: number of parallel threads to run clingo on
     :type pnum: integer
 
+    :param optim: a list containing configuration for optimization algorithm and optionally a bound  [<arg>, <bound>]
+        <arg>: <mode {opt|enum|optN|ignore}>[,<bound>...]
+        opt   : Find optimal model
+        enum  : Find models with costs <= <bound>
+        optN  : Find optimum, then enumerate optimal models
+        ignore: Ignore optimize statements
+        <bound> : Set initial bound for objective function(s)
+    :type optim: list
+
     :returns: results of equivalent class
     :rtype: dictionary
     """
+    if optim is None:
```

Comment:
If we put the default value as a list, we get a warning "Default argument value is mutable"
which I agree with since lists in python are pass-by-reference.
we can change the way I formulated it as a list, but for options like "enum" we need two arguments. the option(string) and a bound (integer)